lib/infrastructure/pigeons/pair_provider.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:cobble/infrastructure/pigeons/pigeons.dart' as pigeon;
+import 'package:hooks_riverpod/all.dart';
+
+class PairCallbacks extends StateNotifier<int> implements pigeon.PairCallbacks {
+  PairCallbacks() : super(null);
+
+  @override
+  void onWatchPairComplete(pigeon.NumberWrapper arg) {
+    state = arg.value;
+  }
+}
+
+/// Stores the address of device you are paired to. Can be null.
+final pairProvider = StateNotifierProvider<PairCallbacks>((ref) {
```

Comment:
I don't think this should be made into a global provider. `onWatchPairComplete` is a single fire-and-forget event. If you save that one globally, it may cause issues when user wants to pair again.

Reply:
StreamProvider, then?

Reply:
Yes

Reply:
Done